### DISCUSSION TITLE
Not able to build with Numpy support on Windows 10

### DISCUSSION THREAD

**Discussion by FeliciaBrisc** [AcademySoftwareFoundation/openvdb#1200] [Q&A] [answered]
Hello all, 
When trying to build with Numpy support I get the error: 
-- Found Python: /VCPKG/vcpkg/installed/x64-windows/include/python3.9) (found suitable version "3.9.7", minimum required is "2.7")
CMake Error at openvdb/openvdb/python/CMakeLists.txt:103 (message):
Could NOT find NumPy: Found unsuitable version ""but required is at least
 "1.14.0" (found C:/VCPKG/vcpkg/installed/x64-windows/include/boost/python)

I previously installed vcpkg and the following dependencies, according to the README instructions 
vcpkg install zlib:x64-windows
vcpkg install blosc:x64-windows
vcpkg install tbb:x64-windows
vcpkg install boost-iostreams:x64-windows
vcpkg install boost-system:x64-windows
vcpkg install boost-any:x64-windows
vcpkg install boost-algorithm:x64-windows
vcpkg install boost-uuid:x64-windows
vcpkg install boost-interprocess:x64-windows
vcpkg install boost-python:x64-windows

Then I built openvdb with: cmake -DCMAKE_TOOLCHAIN_FILE=\VCPKG\vcpkg\\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DPython_NumPy_INCLUDE_DIR=C:/VCPKG/vcpkg/installed/x64-windows/include/boost/python -DOPENVDB_BUILD_PYTHON_MODULE=ON -D USE_NUMPY=ON -A x64 .. 

I also attached the CMake file. 

I also tried to use instead of the VCPKG Python installation my Anaconda Python 3.9.7 installation:

Python_EXECUTABLE:FILEPATH=C:/Users/FB/anaconda3/envs/PYTHON39/python.exe
Python_INCLUDE_DIR=C:/Users/FB/anaconda3/envs/PYTHON39/include
Python_NumPy_INCLUDE_DIR:UNINITIALIZED=C:/Users/FB/anaconda3/envs/PYTHON39/Lib/site-packages/numpy/core/include/numpy

Got however the same error, Could NOT find NumPy: Found unsuitable version ""but required is at least
 "1.14.0" - although the Numpy version is 1.21.2

Many thanks for any help. 

[CMakeCache.txt](https://github.com/AcademySoftwareFoundation/openvdb/files/7322478/CMakeCache.txt)

**Comment by Idclip** [comment, marked as answer]
Looks like CMake isn't setting `Python_NumPy_VERSION` for some reason. Can you try without providing the path for `Python_INCLUDE_DIR=`? If that doesn't work you can probably just set the version yourself to circumvent the issue i.e:`cmake -DPython_NumPy_VERSION="1.21.2" <other args>"`

**Reply by FeliciaBrisc** [reply]
Thank you for you reply @Idclip - unfortunately setting Python_NumPy_VERSION doesn't appear to affect the result- 

If I leave out the Python_NumPy_INCLUDE_DIR then Numpy isn't found: CMake Error at openvdb/openvdb/python/CMakeLists.txt:99 (message):  Could NOT find NumPy (Required is at least version "1.14.0")

If I supply the Python_NumPy_INCLUDE_DIR, Numpy is found, but with no suitable version (inspite of specifying this time Python_NumPy_VERSION). I tried to circumvent this by commenting out in CMakeLists the relevant lines. However, afterwards boost_numpy isn't found: 
-- Found Python: /VCPKG/vcpkg/installed/x64-windows/include/python3.9) (found suitable version "3.9.7", minimum required is "2.7")
-- Found NumPy: C:/VCPKG/vcpkg/installed/x64-windows/include/boost/python (found suitable version "", minimum required is "1.14.0")
-- Found boost_python39
CMake Error at openvdb/openvdb/python/CMakeLists.txt:193 (message):
  Unable to find boost_numpy39, boost_numpy3 or boost_numpy.

The two CMakeFiles for these two cases are attached. It also appears that vcpkg didn't build any boost_numpy library. 

I tried to point to the boost libraries installed in Anaconda, to no avail unfortunately. 
I should also mention that without Numpy, I had a successful build - however I need Numpy for my work. 
[CMakeCache_NoPythonNumpyInclude.txt](https://github.com/AcademySoftwareFoundation/openvdb/files/7330377/CMakeCache_NoPythonNumpyInclude.txt)
[CMakeCache_WithNumpyInclude.txt](https://github.com/AcademySoftwareFoundation/openvdb/files/7330379/CMakeCache_WithNumpyInclude.txt)

**Reply by Idclip** [reply]
Commenting out the check for `Python_NumPy_VERSION` was going to be my next suggestion, hopefully that's all that's needed there! Regarding boost_numpy, as you've noticed, vcpkg doesn't have good support for this. I can point you to the open issue with the vcpkg maintainers where some suggestions have been made:

https://github.com/microsoft/vcpkg/issues/5732

If you instead want to use the version of boost provided by conda you _should_ be able to point CMake to the boost root with `-DBOOST_ROOT=<path>`. It sounds like you tried this but this didn't work? I'd also experiment with setting `-DBoost_DIR=<path/to/conda/boost/cmake>` or `-DBOOST_INCLUDEDIR=<root>/include -DBOOST_LIBRARYDIR=<root>/lib`. Additionally you could try and use CMakes native boost modules by setting `-DBoost_NO_BOOST_CMAKE=ON`.

Let me know how that goes - if it fails I will try and replicate the issue.

**Reply by FeliciaBrisc** [reply]
I got sidetracked by other projects, but I finally managed to finish my installation. Nothing appeared to work when trying to use the boost libraries from conda or from VCPCK.  I then installed separately boost on my system and pointed to it's libraries with -DBOOST_ROOT, -DBOOST_INCLUDEDIR  and -DBOOST_LIBRARYDIR. 
I also had to specify manually the boost compiler parameter in cmake to match the compiler version used to build Boost: 
in Program Files\CMake\share\cmake-3.21\Modules\FindBoost.cmake I modified 'set(_boost_COMPILER ${Boost_COMPILER})' to  'set(_boost_COMPILER "-vc141")'. 
Please mark this question as solved. Many thanks for your help with my installation.  

[CMakeCache.txt](https://github.com/AcademySoftwareFoundation/openvdb/files/8341001/CMakeCache.txt)
